lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3748,6 +3748,10 @@ LogicalResult FIRRTLLowering::visitExpr(ShlPrimOp op) {
 }
 
 LogicalResult FIRRTLLowering::visitExpr(ShrPrimOp op) {
+  // If this is a 0-bit value shifted by any amount, then return a 1-bit zero.
+  if (!op.getInput().getType().getBitWidthOrSentinel())
+    return setLowering(op, getOrCreateIntConstant(1, 0));
```

Comment:
Consider `isZeroBitFIRRTLType(op.getInput().getType())` for consistency with other paths but this is good too :+1:.

Reply:
Since I went to double-check, just noting for myself/others that this is the current shr behavior and not the new FIRRTL 4.0 behavior. (as it should be until we switch over).

Reply:
Yep, this is better, thanks! Will change to this.